fixed mapbase autoalign (#6796)
sunpy/sunpy: changelog/6796.bugfix.rst

> When using ``autoalign=True`` when plotting maps, the result was off by half a pixel as the ``pcolormesh`` call was not supplying the pixel edges explicitly.

Reviewer: ```suggestion
When using ``autoalign=True`` when plotting maps, the result was misaligned by half a pixel.
```
There's no particular benefit here to providing detail

> Fixed a bug in ### sunpy/map/mapbase.py file to supply the pixeledges to the ####pcolormesh() function explicitly while calling pcolormesh() function, so it does not treat integer pixels as pixel edges.

Reviewer 2: ```suggestion
When using ``autoalign=True`` when plotting maps, the result was off by half a pixel as the ``pcolormesh`` call was not supplying the pixel edges explicitly.
```
